pkg/qsfsd/qsfs.go

```
@@ -181,6 +181,52 @@ func (f *QSFS) isMounted(path string) (bool, error) {
 	return false, nil
 }
 
+func (q *QSFS) UpdateMount(wlID string, cfg zos.QuantumSafeFS) (pkg.QSFSInfo, error) {
+	var info pkg.QSFSInfo
+	zstorConfig := setQSFSDefaults(&cfg)
+	networkd := stubs.NewNetworkerStub(q.cl)
+	flistd := stubs.NewFlisterStub(q.cl)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer cancel()
+	netns := networkd.QSFSNamespace(ctx, wlID)
+	ips, _, err := networkd.Addrs(ctx, "ygg0", netns)
+	if err != nil {
+		return info, errors.Wrap(err, "failed to get ygg ip")
+	}
+	if len(ips) != 1 {
+		return info, errors.Wrap(err, "multiple ips found at ygg interface")
+	}
+	yggIP := ips[0]
+	flistPath, err := flistd.Mount(ctx, wlID, qsfsFlist, pkg.MountOptions{
```

I strongly recommend that this instead be a new method (UpdateMount) that takes ONLY the wlID and new size) this method can then only process already mounted flists, to increase(or theoretically decrese) the volume.

My point is, this now gives the impression that an flist, or mount mode is also changible!. The side effect of this method is then not clear no consistent. right ?

An update method on the other hand, is explicit that first, I am doing an update of an already mounted flist, the mount update only supports size, and it works only for rw mounts, hence a ro mount should return an error.